### PR TITLE
Fix rp focus selections in e2e tests needing azure client

### PR DESCRIPTION
**Pull request by charlesakalugwu**
follow-up to #880 

/cc @kargakis @jim-minter

### CODE REVIEW

**Comment by thekad** [comment]
Does this require a change in `./hack/e2e.sh` ?

**Comment by charlesakalugwu** [comment]
@thekad no changes in `./hack/e2e.sh` required with this change

**Comment by kargakis** [comment]
Just reviewed this, it still seems to be an issue. I would expect this PR to fail because `make e2e`'s FOCUS does not include Fake, right?

File: test/clients/azure/client.go [hunk 1]
```diff
@@ -50,12 +59,12 @@ func NewClientFromEnvironment() (*Client, error) {
 	}
 
 	var rpURL string
-	focus := []byte(config.GinkgoConfig.FocusString)
+	focus := config.GinkgoConfig.FocusString
 	switch {
-	case adminRpFocus.Match(focus), fakeRpFocus.Match(focus):
+	case adminRpFocus.match(focus), fakeRpFocus.match(focus):
 		fmt.Println("configuring the fake resource provider")
 		rpURL = realfakerp.StartServer(log, conf, realfakerp.LocalHttpAddr)
```
Comment (kargakis) [inline test/clients/azure/client.go]: Btw, this is wrong, StartServer should ensure there is no fakerp running before trying to start a new one.

Comment (kargakis) [reply test/clients/azure/client.go]: Probably doesn't matter since we don't check the error returned from ListenAndServe.

**Comment by kargakis** [comment]
> Just reviewed this, it still seems to be an issue. I would expect this PR to fail because `make e2e`'s FOCUS does not include Fake, right?

Ok, this does not fail because we don't use the azure client in any of these tests.

**Comment by kargakis** [comment]
> Ok, this does not fail because we don't use the azure client in any of these tests.

I am setting the tags correctly in the second commit in https://github.com/openshift/openshift-azure/pull/831 (had to otherwise my PR fails to pass e2e).

**Pull request merged**